test/parallel/test-http-keepalive-free.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+
+const http = require('http');
+
+const server = http.createServer(common.mustCall((req, res) => {
+  res.end(req.url);
+}));
+server.listen(0, common.mustCall(() => {
+  const agent = http.Agent({ keepAlive: true });
+
+  const req = http
+    .request({
+      port: server.address().port,
+      agent
+    })
+    .on('socket', common.mustCall((socket) => {
+      socket.on('free', common.mustCall());
+    }))
+    .on('response', common.mustCall((res) => {
+      assert.strictEqual(req.destroyed, false);
+      res.on('end', () => {
+        assert.strictEqual(req.destroyed, true);
+        req.destroy();
```

Comment:
I know that in v14.0.0. `req.abort()` points to `req.destroy()`, but I'd add an additional test for `req.abort()` to prevent regressions.